### PR TITLE
caching api response with sessionStorage

### PR DESCRIPTION
**Pull request by Mrkezii**
This prevents the Search route from always making a get request to the api to return the PVC Collection centers.

Here, with the use of ```sessionStorage```, after the initial API call, the results are cached in the session storage, the results are available throughout that session till the tab is closed.

What i noticed earlier was that after i navigate out of the search route, on my return the page makes an API call again and it'll continue making calls whenever you navigate out and come back.

### CODE REVIEW

**Comment by yomete** [comment]
Any reason why sessionStorage was preferred over localStorage?

**Comment by Mrkezii** [comment]
Because sessionStorage is only cached for every active session and i think a fresh api call is needed when a user is visiting the site for a new session.

It can be changed to localStorage if you think that is more beneficial to the end users

**Review by pyjac — COMMENTED** [review]
You might want to revisit the spacing.

File: src/routes/search/index.js [hunk 1]
```diff
@@ -1,517 +1,854 @@
...
+		const sessionStorageRef = JSON.parse(
+			sessionStorage.getItem(`PVCLocations`)
+		);
+		// check if session storage already exists, if not make a get request to the API
+		if (sessionStorageRef) {
+			this.setState({
+				value: query,
+				loading: true,
+				locations: sessionStorageRef
+			});
```
Comment (pyjac) [inline src/routes/search/index.js:118]: You can [return early](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) here...

File: src/routes/search/index.js [hunk 2]
```diff
@@ -1,517 +1,854 @@
...
-      visiblesuggest: false,
-      visibledelete: false,
-      showdisqus: false,
-      currentLocation: null
-    }
+	constructor() {
+		super();
+		this.state = {
+			result: false,
+			value: "",
```
Comment (pyjac) [inline src/routes/search/index.js:85]: Why do you prefer double quotes?

File: src/routes/search/index.js [hunk 3]
```diff
@@ -1,517 +1,854 @@
...
-        this.setState({ result: true })
-        route(`${this.props.path}?query=${this.state.value}`)
-      })
-      .catch(err => {
-        console.error(err)
-      })
-    event.preventDefault()
-  }
+	show(val) {
+		return e => {
```
Comment (pyjac) [inline src/routes/search/index.js:13]: Why are you returning a function here? and for `showsuggest`, `showdeletion`, `showDisqus` and `closeDisqus`

**Review by pyjac — CHANGES_REQUESTED** [review]
Can you pull latest changes on `ErxiaHQ/govote-app@development` and rebase your branch on that? Some of your changes removed the linting fixes earlier merged in.

**Comment by yomete** [comment]
Yeah @Mrkezii You should pull from the development branch. @pyjac already pushed some linting fixes

File: src/routes/search/index.js (hunk 2 above)
Comment (Mrkezii) [reply src/routes/search/index.js:85]: just preference, nothing else 😊

File: src/routes/search/index.js (hunk 3 above)
Comment (Mrkezii) [reply src/routes/search/index.js:13]: haven't touched that side of the code, still the original stuff.
@yomete can answer this better

**Pull request closed without merging**